seacliff/rubin_sky_wcs.py

```diff
@@ -0,0 +1,132 @@
+import numpy as np
+
+import lsst.geom
+import lsst.afw.geom
+import galsim
+from galsim.wcs import CelestialWCS
+
+
+class RubinSkyWCS(CelestialWCS):
+    """A galsim-compatible wrapper of the Rubin SkyWcs class.
+
+    NOTE: GalSim uses FITS 1-based indexing whereas Rubin uses 0-based indexing. This
+    WCS class uses the galsim FITS conventions and explicitly translates via offsets of
+    -1 or 1 as needed.
+
+    Parameters
+    ----------
+    wcs : lsst.afw.geom.SkyWcs
+        The Rubin Sky WCS to be wrapped by this class.
+    origin : PositionD or PositionI or None, optional
+        If not None, the origin position of the image coordinate system. Note that
+        the conversion from 1-based to 0-based pixel indexing is **always** done.
+    """
+
+    # TODO: write a builder function for the config parser and settle on final API
+    _req_params = {"wcs": lsst.afw.geom.SkyWcs}
```

Comment:
TBH, if you're going to make an `lsst.aft.geom.SkyWcs` object be required, there's no point in having these `_req_params` (or `_opt_params`) at all.  They just allow for a quick description of how this could be read from a config dict without requiring any custom functions.  That mechanism doesn't know what a `SkyWcs` is, so this is useless.

Once we want this to be a config item, we'll have to write a custom loader that knows how to make a `SkyWcs`.